Add initial GSoC 2020 ideas list

vnen opened the pull request:
To be refined we new ideas and description.

vnen committed:
Add initial GSoC 2020 ideas list

Calinou commented on GSoC/GSoC-2020.md:
```diff
@@ -0,0 +1,58 @@
+* Integration with VCS (such as Git, SVN, Mercurial, etc.)
+* Resolution of conflict when a new resource is added that requires new files (i.e. someone adds a texture that other people don't have. Maybe only allow new references to files that already exist in the VCS for simplicty, or provide warnings/placeholders when a client doesn't have the file).
+
+## Improvement in i18n for games
```
I think it'd be interesting to mention support for [pluralization](https://github.com/godotengine/godot/issues/31432) here.

Also, is the performance difference between PO and MO files noticeable on a medium-sized project? If not, it may be difficult to justify the added complexity.

vnen replied:
Pluralization is mentioned already, I'll add a link to the issue.

> Also, is the performance difference between PO and MO files noticeable on a medium-sized project (say, 2,000 strings)? If not, it may be difficult to justify the added complexity. That doesn't mean it can't be added; I just consider it to be lower-priority compared to features like pluralization.

MO should be quite smaller, which is already a load optimization. Also, since it's binary, I assume it's actually easier to parse because it needs a rigid structure. I do agree that translation features are more important than formats, but they usually go hand-in-hand too.

Maybe we don't even want gettext. If there's a better more modern format which supports more i18n features we may actually want it instead.

Calinou replied:
> Maybe we don't even want gettext. If there's a better more modern format which supports more i18n features we may actually want it instead.

I think the main established alternative out there is XLIFF. It tends to be more common in proprietary software circles and is more verbose overall – it's not fun to edit with a text editor. So, I think we can keep gettext for now.

vnen commented:
Added some new ideas, some discussed at the Godot Sprint, some taken from GIPs. Removed the multiplayer scene editor, since we don't know if we really want that at this point (also probably too much for a summer internship).

I'm gonna be merging this very soon, since we need for the GSoC application. I'll leave it open to review for typos and stuff.
